lambdas/indexer/app.py

```diff
@@ -334,12 +280,12 @@ class DocumentTally:
     attempts: int
 
     @classmethod
-    def from_message(cls, message) -> 'DocumentTally':
-        body = json.loads(message.body)
+    def from_sqs_record(cls, record: Mapping[str, JSON]) -> 'DocumentTally':
```

Comment:
The type hint is incorrect and I would excuse a mistake like that if it weren't for the fact that PyCharm's type checker warns about it. I would check my PyCharm settings to ensure you are made aware of type checker warnings in PyCharm.

The reason I am such a stickler for type hints is that type hints are extremely useful in declaring intent AND and having that declaration be partially checked by tooling.

`JSON` is `Mapping`. So `Mapping[str, JSON]` implies that the function expects a dictionary where the values are dictionaries. But the very first line passes such a value to a function that expects a string (`json.loads`). This is obviously wrong.

Either way, I want to simplify this further. See my diff.